cli/cmd/aeon.go

```diff
@@ -145,3 +159,85 @@ func internalAeonConnect(cmdCtx *cmdcontext.CmdCtx, args []string) error {
 	}
 	return nil
 }
+
+func readConfigFile(args []string) error {
+	var (
+		configPath = args[0]
+		instance   = args[1]
+	)
+
+	f, err := os.ReadFile(configPath)
+	if err != nil {
+		return err
+	}
+
+	pb := cluster.NewYamlCollector(f)
+	config, err := pb.Collect()
+	if err != nil {
+		return err
+	}
+
+	clusterConfig, err := cluster.MakeClusterConfig(config)
+	if err != nil {
+		return err
+	}
+
+	result := cluster.Instantiate(clusterConfig, instance)
+
+	// Get SSL connection.
+	dataSsl := []string{"roles_cfg", "aeon.grpc", "advertise"}
+	data, err := result.Get(dataSsl)
+	if err != nil {
+		return err
+	}
+
+	var advertise aeoncmd.Advertise
+	err = mapstructure.Decode(data, &advertise)
+	if err != nil {
+		return err
+	}
+
+	if advertise.Uri == "" {
+		return errors.New("invalid connection url")
+	}
+
+	re := regexp.MustCompile("^https?://")
+	cleanedURL := re.ReplaceAllString(advertise.Uri, "")
+
+	connectCtx.Network, connectCtx.Address = libconnect.ParseBaseURI(cleanedURL)
+
+	if (advertise.Params.Transport != "ssl") && (advertise.Params.Transport != "plain") {
+		return errors.New("transport must be ssl or plain")
+	}
+
+	if advertise.Params.Transport == "ssl" {
+		connectCtx.Transport = aeoncmd.TransportSsl
+
+		if advertise.Params.KeyFile == "" || advertise.Params.CertFile == "" {
+			return errors.New("files Key and Cert must be specified both")
+		}
+
+		if util.IsRegularFile(advertise.Params.KeyFile) {
+			return fmt.Errorf("not valid path to a private SSL key file=%q",
+				advertise.Params.KeyFile)
+		}
+		if util.IsRegularFile(advertise.Params.CertFile) {
+			return fmt.Errorf("not valid path to an SSL certificate file=%q",
+				advertise.Params.CertFile)
+		}
+		if util.IsRegularFile(advertise.Params.CaFile) {
+			return fmt.Errorf("not valid path to trusted certificate authorities (CA) file=%q",
+				advertise.Params.CaFile)
+		}
+
+		connectCtx.Ssl = aeoncmd.Ssl{
+			KeyFile:  advertise.Params.KeyFile,
+			CertFile: advertise.Params.CertFile,
+			CaFile:   advertise.Params.CaFile,
+		}
+	} else {
```

Comment:
The idea here: don't change user flags if a transport is not set in the config.
```suggestion
	} else if advertise.Params.Transport == aeoncmd.TransportPlain {
```

Reply:
Done